move ProcessGroupGlooTest to gtest

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32138 use gtest asserts in ProcessGroupGlooTest instead of other checks
* #32134 [flaky-tests] fix testSend and testRecv in ProcessGroupGlooTest
* **#32133 move ProcessGroupGlooTest to gtest**

We should do this to better debug individual tests in this file.

Differential Revision: [D19375479](https://our.internmc.facebook.com/intern/diff/D19375479/)

rohan-varma committed:
move ProcessGroupGlooTest to gtest

We should do this to better debug the test.

Differential Revision: [D19375479](https://our.internmc.facebook.com/intern/diff/D19375479/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 149d87bf:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=149d87bf8137eff3cc6a8c2a0284c93eb40fc7e0).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4225369) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=84468260) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=84468260))
<details>
<summary>
<code>Jan 13 22:13:55 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 13 22:13:55 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 13 22:13:55 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 13 22:13:55 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 13 22:13:55 1 out of 1 hunk FAILED 
Jan 13 22:13:55 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 13 22:13:55 Traceback (most recent call last): 
Jan 13 22:13:55   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 13 22:13:55     patch_repo(args) 
Jan 13 22:13:55   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 13 22:13:55     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 13 22:13:55 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 13 22:13:55 =================== sccache compilation log =================== 
Jan 13 22:13:55 + cleanup 
Jan 13 22:13:55 + retcode=1 
Jan 13 22:13:55 + set +x 
Jan 13 22:13:55 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 13 22:13:55 Compile requests                 1 
Jan 13 22:13:55 Compile requests executed        0 
Jan 13 22:13:55 Cache hits                       0 
Jan 13 22:13:55 Cache misses                     0 
Jan 13 22:13:55 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

rohan-varma committed:
Update on "move ProcessGroupGlooTest to gtest"


We should do this to better debug individual tests in this file.

Differential Revision: [D19375479](https://our.internmc.facebook.com/intern/diff/D19375479/)

[ghstack-poisoned]

rohan-varma committed:
Update on "move ProcessGroupGlooTest to gtest"


We should do this to better debug individual tests in this file.

Differential Revision: [D19375479](https://our.internmc.facebook.com/intern/diff/D19375479/)

[ghstack-poisoned]

mrshenli reviewed (commented):
This LGTM. 

@yf225 is there anything we need to be aware of when making changes to C++ tests?

mrshenli commented on torch/lib/c10d/test/ProcessGroupGlooTest.cpp:
```diff
@@ -381,6 +383,7 @@ int main(int argc, char** argv) {
     }
   }
 
+  // test SIGKILL
```
Should this be a separate test case?

rohan-varma replied:
I will move this in the 3rd PR in this stack.

mrshenli reviewed (approved):
I guess as these tests live in a self-contained location other than `test/cpp`, they didn't keep up the same standard as other cpp tests. This PR would help make them more consistent with other cpp tests. Thanks!

rohan-varma committed:
Update on "move ProcessGroupGlooTest to gtest"


We should do this to better debug individual tests in this file.

Differential Revision: [D19375479](https://our.internmc.facebook.com/intern/diff/D19375479/)

[ghstack-poisoned]
